resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/k8s/features/DriverServiceFeatureStepSuite.scala

```diff
@@ -92,25 +91,22 @@ class DriverServiceFeatureStepSuite extends SparkFunSuite {
   }
 
   test("Long prefixes should switch to using a generated name.") {
-    val clock = new ManualClock()
-    clock.setTime(10000)
     val sparkConf = new SparkConf(false)
       .set(KUBERNETES_NAMESPACE, "my-namespace")
-    val configurationStep = new DriverServiceFeatureStep(
-      KubernetesTestConf.createDriverConf(
-        sparkConf = sparkConf,
-        resourceNamePrefix = Some(LONG_RESOURCE_NAME_PREFIX),
-        labels = DRIVER_LABELS),
-      clock)
+    val kconf = KubernetesTestConf.createDriverConf(
+      sparkConf = sparkConf,
+      resourceNamePrefix = Some(LONG_RESOURCE_NAME_PREFIX),
+      labels = DRIVER_LABELS)
+    val configurationStep = new DriverServiceFeatureStep(kconf)
+
     val driverService = configurationStep
       .getAdditionalKubernetesResources()
       .head
       .asInstanceOf[Service]
-    val expectedServiceName = s"spark-10000${DriverServiceFeatureStep.DRIVER_SVC_POSTFIX}"
-    assert(driverService.getMetadata.getName === expectedServiceName)
-    val expectedHostName = s"$expectedServiceName.my-namespace.svc"
+    assert(!driverService.getMetadata.getName.startsWith(kconf.resourceNamePrefix))
```

Comment:
Hi, @vanzin . Can we check the actual pattern you introduced here? Maybe, we can use `.startWith` and `endWith`.
```
val shorterServiceName = s"spark-$randomServiceId$DRIVER_SVC_POSTFIX"
```
Negative assertions(at 106 and 109) looks insufficient to check the new logic. If something happens on the patterns, this test doesn't catch the change at all. Previously, we check the name pattern [exactly](https://github.com/apache/spark/pull/23805/files#diff-0d7a047564dda7b7bdbb656040c0600bL110).

Reply:
> If something happens on the patterns, this test doesn't catch the change at all

I'm not sure what you mean by that.

It's not possible to have a positive test. The whole point of the change is to introduce randomness to the identifier being generated.

The tests set the resource prefix to something that does not start with `spark-`, which is the prefix that is used in the auto-generated path. So checking that the final identifier does not start with the requested prefix is enough to test that the change is occurring.

Reply:
The randomness is not about the whole string. It's only about a part of it. Specifically, this PR changes from `spark-10000-driver-svc` to `spark-b805936936efbcd5-driver-svc`.
> The tests set the resource prefix to something that does not start with `spark-`, which is the prefix that is used in the auto-generated path.

Previously, the randomness is determined by time, `10000`. Now, it has a better string, `b805936936efbcd5`.

What I asked is adding the following two lines as a positive test case.
```scala
     assert(!driverService.getMetadata.getName.startsWith(kconf.resourceNamePrefix))
+    assert(driverService.getMetadata.getName.startsWith("spark-"))
+    assert(driverService.getMetadata.getName.endsWith("driver-svc"))
```

The other stuff of this PR looks good.

Reply:
I totally agree with you. So, I mentioned like `I know this test case is focusing only on the long name case`.
> So checking that the final identifier does not start with the requested prefix is enough to test that the change is occurring.

There was no other test case to check this new pattern. So, I asked to keep some coverage.

Reply:
What new pattern? The generated service name? Why is it interesting to test it?

The test checks that the long name is not used. It doesn't matter what name is used as long as it's not the long name. And that's already achieved by the test. As you've said yourself.

Reply:
I changed the test to check uniqueness. But checking that the generated name follows a specific pattern is a rather pointless test.